packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -499,6 +500,35 @@ export const activate = async (extensionContext: vscode.ExtensionContext) => {
 
   void activateTracker.markActivationStop();
   MetricsReporter.extensionPackStatus();
+
+  console.log('DOING TRACE FLAG STUFF');
+
+  const connection = await WorkspaceContext.getInstance().getConnection();
+
+  // If an expired TraceFlag exists, delete it
+  const traceFlags = await connection.tooling.query(
+    "SELECT Id, ExpirationDate FROM TraceFlag WHERE LogType = 'DEVELOPER_LOG'"
+  );
+  console.log(JSON.stringify(traceFlags, null, 2));
+  const currentTime = new Date();
+  const expiredTraceFlagExists = traceFlags.records.filter(
+    (flag: any) => flag.ExpirationDate && new Date(flag.ExpirationDate) < currentTime
+  ).length > 0;
+
+  if (expiredTraceFlagExists) {
+    const traceFlagId = typeof traceFlags.records[0].Id === 'string'
+      ? traceFlags.records[0].Id
+      : '';
+    await connection.tooling.delete('TraceFlag', traceFlagId);
```

Comment:
what happens when it tries to run `delete` with an empty string?  isn't that going to throw?

Another way to say it would be 
`const [expiredTraceFlag] = traceFlags.records.filter(
    (flag: any) => flag.ExpirationDate && new Date(flag.ExpirationDate) < currentTime
  )`.

Then if there is a record there, it'll be expiredTraceFlag and otherwise expiredTraceFlag is undefined.  And only try to delete if there is one (and grab its ID)